diff --git a/node_modules/jmdict-simplified-node/index.js b/node_modules/jmdict-simplified-node/index.js
index b50ce2d..0bd0b48 100644
--- a/node_modules/jmdict-simplified-node/index.js
+++ b/node_modules/jmdict-simplified-node/index.js
@@ -22,6 +22,7 @@ __export(require("./interfaces"));
 function setup(DBNAME, filename = '', verbose = false) {
     return __awaiter(this, void 0, void 0, function* () {
         const db = levelup_1.default(leveldown_1.default(DBNAME));
+        try {
         try {
             const opt = { asBuffer: false };
             const [dictDate, version] = yield Promise.all([db.get('raw/dictDate', opt), db.get('raw/version', opt)]);
@@ -71,6 +72,8 @@ function setup(DBNAME, filename = '', verbose = false) {
         }
         if (batch.length) {
             yield db.batch(batch);
+        }} catch (e) {
+          db.close()
         }
         return { db, dictDate: raw.dictDate, version: raw.version };
     });
